src/main/java/uk/ac/ebi/embl/converter/utils/ConversionEntry.java

```diff
@@ -10,30 +10,59 @@
  */
 package uk.ac.ebi.embl.converter.utils;
 
+import io.vavr.Tuple2;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 public class ConversionEntry {
     String sOID;
     String sOTerm;
     String definition;
     String feature;
-    String qualifier1;
-    String qualifier2;
+    Map<String, String> qualifiers;
 
     ConversionEntry(String[] tokens) {
         this.sOID = tokens[0];
         this.sOTerm = tokens[1];
         this.definition = tokens[2];
         this.feature = tokens[3];
-        for (String token : tokens)
-            if (tokens.length > 4) this.qualifier1 = tokens[4].equalsIgnoreCase("null") ? null : tokens[4];
-        if (tokens.length > 5) this.qualifier2 = tokens[5].equalsIgnoreCase("null") ? null : tokens[5];
+        this.qualifiers = new HashMap<>();
+        List<String> remainingTokens = Arrays.stream(tokens).skip(4).toList();
+        for (String token : remainingTokens) {
+            Tuple2<String, String> parsed = parseQualifier(token);
+            if (parsed != null) {
+                qualifiers.put(parsed._1, parsed._2);
+            }
+        }
     }
 
-    public String getQualifier1() {
-        return qualifier1;
+    private Tuple2<String, String> parseQualifier(String str) {
+        String[] parts = str.split("=");
+        String key;
+        String value;
+        switch (parts.length) {
+            case 2:
+                key = parts[0];
+                value = parts[1].substring(1, parts[1].length() - 1);
+                break;
+            case 1:
+                key = parts[0];
+                value = "true";
+                break;
+            default:
+                throw new RuntimeException("Invalid qualifier format: " + str);
+        }
+        // Remove starting / character on qualifiers
+        if (key.startsWith("/")) {
+            key = key.substring(1);
+        }
+        return new Tuple2<>(key, value);
```

Comment:
Not sure about the use of Tuple here, we could use Map instead
Please use this refactored code:
```
private Map<String, String> parseQualifier(String str) {
        String[] parts = str.split("=");
        String key = parts[0].replaceFirst("/","");
        String value;
        switch (parts.length) {
            case 2 -> value = parts[1].replaceAll("^\"|\"$", ""); // Remove surrounding double quotes
            case 1 -> value = "true";
            default -> throw new RuntimeException("Invalid qualifier format: " + str);
        }
        return Map.of(key, value);
    }
```
**Please add unit test to parseQualifier**